File: src/com/gmail/marcosav2010/logger/ILog.java
package com.gmail.marcosav2010.logger;

import com.gmail.marcosav2010.logger.Logger.VerboseLevel;

public interface ILog {

    void log(Object o);

    void log(Object o, VerboseLevel level);

    void log(Throwable t);

    void log(Throwable t, String msg);
}

File: core/src/main/java/com/gmail/marcosav2010/logger/ILog.kt
package com.gmail.marcosav2010.logger

import com.gmail.marcosav2010.logger.Logger.VerboseLevel

interface ILog {

    fun log(o: Any?)
    fun log(o: Any?, level: VerboseLevel)
    fun log(t: Throwable)
    fun log(t: Throwable, msg: String?)
}